_scripts/ProcessLocalesPlugin.js

```diff
@@ -37,66 +38,88 @@ class ProcessLocalesPlugin {
-              let output = JSON.stringify(data)
+            this.removeEmptyValues(data)
 
-              if (this.compress) {
-                filename += '.br'
-                output = await this.compressLocale(output)
-              }
+            let filename = `${this.outputDir}/${locale}.json`
+            let output = JSON.stringify(data)
 
-              let source = new RawSource(output)
+            if (this.compress) {
+              filename += '.br'
+              output = await this.compressLocale(output)
+            }
 
-              if (IS_DEV_SERVER) {
-                source = new CachedSource(source)
-                this.cache.push({ filename, source })
-              }
+            let source = new RawSource(output)
 
-              compilation.emitAsset(filename, source, { minimized: true })
+            if (IS_DEV_SERVER) {
+              source = new CachedSource(source)
+              this.cache[locale] = { filename, source, mtimeMs }
+            }
 
-              resolve()
-            }))
-          }
+            compilation.emitAsset(filename, source, { minimized: true })
 
-          await Promise.all(promises)
+            resolve()
+          }))
 
           if (IS_DEV_SERVER) {
             // we don't need the unmodified sources anymore, as we use the cache `this.cache`
             // so we can clear this to free some memory
-            delete this.locales
+            delete localeEntry.data
           }
-        }
+        })
+
+        await Promise.all(promises)
       })
     })
   }
 
-  loadLocales() {
-    const activeLocales = JSON.parse(readFileSync(`${this.inputDir}/activeLocales.json`))
+  loadLocales(loadModifiedFilesOnly = false) {
+    if (this.activeLocales.length === 0) {
+      this.activeLocales = JSON.parse(readFileSync(`${this.inputDir}/activeLocales.json`))
+    }
 
-    for (const locale of activeLocales) {
-      const contents = readFileSync(`${this.inputDir}/${locale}.yaml`, 'utf-8')
+    for (const locale of this.activeLocales) {
+      const filePath = `${this.inputDir}/${locale}.yaml`
+      // Cannot use `mtime` since values never equal
+      const mtimeMsFromStats = statSync(filePath).mtimeMs
+      if (loadModifiedFilesOnly) {
+        // Skip reading files where mtime (modified time) same as last read
+        // (stored in mtime)
+        const existingMtime = this.locales[locale]?.mtimeMs
+        if (existingMtime != null && existingMtime === mtimeMsFromStats) {
+          continue
+        }
+      }
+      const contents = readFileSync(filePath, 'utf-8')
       const data = loadYaml(contents)
+      this.locales[locale] = { locale, data, mtimeMs: mtimeMsFromStats }
 
-      this.localeNames.push(data['Locale Name'] ?? locale)
-      this.locales.push({ locale, data })
+      const localeName = data['Locale Name'] ?? locale
+      if (!this.localeNames.includes(localeName)) { this.localeNames.push(localeName) }
```

Comment:
`this.localeNames` is read once when webpack loads the `renderer.config.js` file and passed to the `DefinePlugin`, so you can change this to:
```suggestion
      if (!loadModifiedFilesOnly) {
        this.localeNames.push(localeName)
      }
```